### PR TITLE
Make Player Stats Retrievable Within Lua

### PR DESCRIPTION
**Pull request by USBMenus**
lets the user retrieve other players rank and rp within lua
adds 2 network commands
get_player_rank()
get_player_rp()

### CODE REVIEW

**Comment by gir489returns** [comment]
Is this really necessary when you can just read the globals?

**Comment by maybegreat48** [comment]
You'll have to update the globals if you use them directly. A menu update is all that's required with this one (feel free to close again if you disagree)

**Comment by xiaoxiao921** [comment]
That global is thoroughly used in the c++ menu code already so might aswell expose it

**Review by xiaoxiao921 — CHANGES_REQUESTED** [review]
```cpp
// Lua API: Function
// Table: network
// Name: get_player_rank
// Param: player_id: integer: id of the player
// Returns: Rank of the player if the player id is valid, nil otherwise.
static sol::object get_player_rank(int player_id, sol::this_state state_)
{
	if (big::g_player_service->get_by_id(player_id))
	{
		auto& stats = big::scr_globals::gpbd_fm_1.as<GPBD_FM*>()->Entries[player_id].PlayerStats;
		return sol::make_object(state_.L, stats.Rank);
	}

	return sol::nil;
}
```

Let's make it a bit cleaner. Do the same for the other function please.

**Comment by gir489returns** [comment]
> ```c++
> // Lua API: Function
> // Table: network
> // Name: get_player_rank
> // Param: player_id: integer: id of the player
> // Returns: Rank of the player if the player id is valid, nil otherwise.
> static sol::object get_player_rank(int player_id, sol::this_state state_)
> {
> 	if (big::g_player_service->get_by_id(player_id))
> 	{
> 		auto& stats = big::scr_globals::gpbd_fm_1.as<GPBD_FM*>()->Entries[player_id].PlayerStats;
> 		return sol::make_object(state_.L, stats.Rank);
> 	}
> 
> 	return sol::nil;
> }
> ```
> 
> Let's make it a bit cleaner. Do the same for the other function please.

This is inconconsistent with the other functions in the class.

get_flagged_modder_reason returns a blank string, and get_selected_database_player_rockstar_id returns -1. If we're going to change the incoming, we should change them for all, as all the functions should behave the same. Not sure why get_player_rank needs to return an object? What other properties besides the rank value as an integer does the callee need?

File: src/lua/bindings/network.cpp [hunk 1]
```diff
@@ -227,6 +259,8 @@ namespace lua::network
 
 		auto ns = state["network"].get_or_create<sol::table>();
 
+		ns["get_player_rank"]                          = get_player_rank;
+		ns["get_player_rp"]                            = get_player_rp;
```
Comment (gir489returns) [inline src/lua/bindings/network.cpp:263]: Can you put this in chronological order that they appear in the class?

**Review by gir489returns — CHANGES_REQUESTED** [review]

**Comment by xiaoxiao921** [comment]
> > ```c++
> > // Lua API: Function
> > // Table: network
> > // Name: get_player_rank
> > // Param: player_id: integer: id of the player
> > // Returns: Rank of the player if the player id is valid, nil otherwise.
> > static sol::object get_player_rank(int player_id, sol::this_state state_)
> > {
> > 	if (big::g_player_service->get_by_id(player_id))
> > 	{
> > 		auto& stats = big::scr_globals::gpbd_fm_1.as<GPBD_FM*>()->Entries[player_id].PlayerStats;
> > 		return sol::make_object(state_.L, stats.Rank);
> > 	}
> > 
> > 	return sol::nil;
> > }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Let's make it a bit cleaner. Do the same for the other function please.
> 
> This is inconconsistent with the other classes in the Lua exporter. I think we should have an all or nothing paradigm. Either all of them use Sol objects, or none of them do.

Hmm not wrong, never realized that, well I guess that's whatever then

**Comment by gir489returns** [comment]
> Hmm not wrong, never realized that, well I guess that's whatever then

We should move that to an epic(Yim calls them Chores?), but for this PR, I think what he has is fine, bar a few nitpicks. I really hate that god damn clang formatter, but I'm screaming into the void at this point with it for something that's superfical.

**Comment by Yimura** [comment]
So is this fine?

**Comment by xiaoxiao921** [comment]
The doc left to be desired and it's not generated. (Python script not ran). Other than that yeah it's ok

File: src/lua/bindings/network.cpp (hunk 1 above)
Comment (USBMenus) [reply src/lua/bindings/network.cpp:263]: aren't they already in chronological order?

Comment (USBMenus) [reply src/lua/bindings/network.cpp]: done

**Review by gir489returns — APPROVED** [review]

**Review by xiaoxiao921 — APPROVED** [review]

**Pull request merged**